server/datastore/mysql/software.go

```diff
@@ -4164,6 +4165,16 @@ func (ds *Datastore) ListHostSoftware(ctx context.Context, host *fleet.Host, opt
 			policiesBySoftwareTitleId[p.TitleID] = append(policiesBySoftwareTitleId[p.TitleID], p)
 		}
 
+		icons, err := ds.GetSoftwareIconsByTeamAndTitleIds(ctx, teamID, softwareTitleIds)
+		if err != nil {
+			return nil, nil, ctxerr.Wrap(ctx, err, "get software icons by team and title IDs")
+		}
+
+		iconsBySoftwareTitleID := make(map[uint]fleet.SoftwareTitleIcon, len(icons))
+		for _, icon := range icons {
+			iconsBySoftwareTitleID[icon.SoftwareTitleID] = icon
+		}
```

Comment:
Would it make more sense here to return the icons pre-mapped similar to what we do for VPP apps?